grpc/src/builder.rs

```diff
@@ -0,0 +1,188 @@
+                .collect::<Result<Vec<_>, GrpcClientBuilderCertError>>()?,
+        );
+
+        Ok(Self {
+            connection: (endpoint, tls_config),
+            ..self
+        })
+    }
+
+    /// Enables the webpki roots.
+    #[cfg(feature = "tls-webpki-roots")]
+    pub fn with_webpki_roots(self) -> Self {
+        let roots = webpki_roots::TLS_SERVER_ROOTS.iter().cloned();
+        let (endpoint, tls_config) = self.connection;
+        Self {
+            connection: (endpoint, tls_config.trust_anchors(roots)),
+            ..self
+        }
+    }
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+impl<S> GrpcClientBuilder<(Endpoint, ClientTlsConfig), S> {
+    /// Connects to the endpoint, applying tls configuration
+    pub fn connect(self) -> Result<GrpcClientBuilder<Channel, S>> {
+        let (endpoint, tls_config) = self.connection;
+        let connection = endpoint.tls_config(tls_config)?.connect_lazy();
+        Ok(GrpcClientBuilder {
+            connection,
+            signer: self.signer,
+            account_pubkey: self.account_pubkey,
+        })
+    }
+}
+
+#[cfg(target_arch = "wasm32")]
+impl GrpcClientBuilder<tonic_web_wasm_client::Client, ()> {
+    /// Create a new client connected to the given `url` with default
+    /// settings of [`tonic_web_wasm_client::Client`].
+    pub fn with_grpcweb_url(url: impl Into<String>) -> Self {
+        let connection = tonic_web_wasm_client::Client::new(url.into());
+        GrpcClientBuilder {
+            connection,
+            signer: (),
+            account_pubkey: None,
+        }
+    }
+}
+
+impl<T> GrpcClientBuilder<T, ()> {
+    /// Create a gRPC client builder using provided prepared transport
+    pub fn with_transport(transport: T) -> Self {
+        Self {
+            connection: transport,
+            signer: (),
+            account_pubkey: None,
+        }
+    }
+
+    /// Add signer and a public key
+    pub fn with_pubkey_and_signer<S>(
+        self,
+        account_pubkey: VerifyingKey,
+        signer: S,
+    ) -> GrpcClientBuilder<T, S> {
+        GrpcClientBuilder {
+            connection: self.connection,
+            signer,
+            account_pubkey: Some(account_pubkey),
+        }
+    }
+
+    /// Add signer and associated public key
+    pub fn with_signer_keypair<S>(self, signer: S) -> GrpcClientBuilder<T, S>
+    where
+        S: Keypair<VerifyingKey = VerifyingKey>,
+    {
+        let account_pubkey = Some(signer.verifying_key());
+        GrpcClientBuilder {
+            connection: self.connection,
+            signer,
+            account_pubkey,
+        }
+    }
+}
+
+impl<T> GrpcClientBuilder<T, ()>
+where
+    T: GrpcService<BoxBody> + Clone,
+    T::ResponseBody: Body<Data = Bytes> + Send + 'static,
+    <T::ResponseBody as Body>::Error: Into<StdError> + Send,
+{
+    /// Build [`GrpcClient`]
+    pub fn build_client(self) -> GrpcClient<T> {
+        GrpcClient::new(self.connection)
+    }
```

Comment:
I think I would rename `build` to `connect`, and if we were to keep both grpc and tx client, I'd call them `connect` and `connect_read_only` /  `connect_ro`